tls/README.md

```diff
@@ -0,0 +1,287 @@
+              "ValidityDuration": "10y",
+              "Usages": [
+                "KeyUsageDigitalSignature",
+                "KeyUsageKeyEncipherment",
+                "KeyUsageCertSign"
+              ],
+              "ExtendedUsages": []
+            }
+          ]
+        },
+        "Status": {
+          "Errors": null
+        }
+...
+  CertKeyPairs": {
+    "Items": [
+      {
+        "LogicalName": "",
+        "Description": "",
+        "Name": "metrics.openshift-apiserver-operator.svc::3074380002740555304",
+        "Spec": {
+          "SecretLocations": [
+            {
+              "Namespace": "openshift-apiserver-operator",
+              "Name": "openshift-apiserver-operator-serving-cert"
+            }
+          ],
+          "OnDiskLocations": null,
+          "CertMetadata": {
+            "CertIdentifier": {
+              "CommonName": "metrics.openshift-apiserver-operator.svc",
+              "SerialNumber": "3074380002740555304",
+              "Issuer": {
+                "CommonName": "openshift-service-serving-signer@1701354509",
+                "SerialNumber": "",
+                "Issuer": null
+              }
+            },
+            "SignatureAlgorithm": "SHA256-RSA",
+            "PublicKeyAlgorithm": "RSA",
+            "PublicKeyBitSize": "2048 bit",
+            "ValidityDuration": "2y",
+            "Usages": [
+              "KeyUsageDigitalSignature",
+              "KeyUsageKeyEncipherment"
+            ],
+            "ExtendedUsages": [
+              "ExtKeyUsageServerAuth"
+            ]
+          },
+          "Details": {
+            "CertType": "ServingCertDetails",
+            "SignerDetails": null,
+            "ServingCertDetails": {
+              "DNSNames": [
+                "metrics.openshift-apiserver-operator.svc",
+                "metrics.openshift-apiserver-operator.svc.cluster.local"
+              ],
+              "IPAddresses": null
+            },
+            "ClientCertDetails": null
+          }
+        },
+        "Status": {
+          "Errors": null
+        }
+      },
+```
+
+This stores the following info:
+* all secrets / configmaps which are TLS artifacts along with their metadata
+* parses them and stores metadata
+* removes locations that are versioned or hashed copies of existing TLS artifacts
+* deduplicates them by content, so copies of TLS artifacts are grouped by location
+
+## Certificate metadata
+
+TLS artifact contents may however be insufficient - i.e. it's not clear which OpenShift is responsible 
+for the TLS artifacts lifecycle. To do that the TLS artifacts should have additional annotations.
+
+Recently we've added the `openshift.io/owning-component: foobar` annotation to (almost) all TLS artifacts
+so that issues related to this TLS artifact would be routed to a proper location. For instance, 
+problems with `service-ca` TLS artifacts are filed in Jira for "Networking / cluster-network-operator" component 
```

Comment:
```suggestion
problems with `service-ca` TLS artifacts are filed in Jira for "service-ca" component 
```

or perhaps you intended to point to the proxy-cas?

Reply:
Yeah, I meant proxy CAs